forem/forem
# Make Consumer Apps dictate aasa results

## What type of PR is this? (check all applicable)


## Description

This PR makes [Apple Application Site Association file (aasa)](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html) automatically generated by the Consumer Apps in the DB. 

These Consumer Apps can be individually managed by each Forem creator, which means creators would now have an option to add iOS Universal Link support for 3rd party apps.

## QA Instructions, Screenshots, Recordings

1. Boot the app and login with Admin user
2. Visit `/admin/apps/consumer_apps`
3. Click on "New Consumer App"
4. Play around with the form
   - Switching between `iOS`/`Android` should display/hide the "Team ID" option (only visible for iOS Consumer Apps)
   - Only required values are "App Bundle" and "Platform"

https://user-images.githubusercontent.com/6045239/122968096-6923ec00-d348-11eb-9e1a-1631eac5867c.mov

### UI accessibility concerns?

None - We're just adding a new field to an existing form, but I'm be open to implement any suggestion in this regard

## Added/updated tests?


## [Forem core team only] How will this change be communicated?

      or in a [forem.dev](http://forem.dev) post

## [optional] Are there any post deployment tasks we need to perform?

DEV admin should create a ConsumerApp for the legacy DEV iOS app in order to preserve that backward compatibility

## [optional] What gif best describes this PR or how it makes you feel?

![Building blocks smiley face](https://media.giphy.com/media/Zxo0sziR6WDDOSd0Wy/giphy.gif)
